restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java

```diff
@@ -2289,6 +2225,9 @@ public void response(final Object message) throws IOException {
                 SipServletRequest invite = response.getRequest();
                 challengeRequest.setContent(invite.getContent(), invite.getContentType());
                 invite = challengeRequest;
+
+                Map<String,ArrayList<String>> extensionHeaders = (Map<String,ArrayList<String>>)linkedB2BUASession.getAttribute(B2BUAHelper.EXTENSION_HEADERS);
+                B2BUAHelper.addHeadersToMessage(challengeRequest, extensionHeaders, sipFactory);
```

Comment:
i take this is actually the only change to fix the issue, is it not the same condition as before?

Reply:
Previously we never modified the headers for this B2BUA/challenge request case.